sudachi/src/dic/category_type.rs

```diff
@@ -14,76 +14,154 @@
  * limitations under the License.
  */
 
-use std::collections::HashSet;
+use crate::error::SudachiError;
+use bitflags::bitflags;
 use std::str::FromStr;
 
-use crate::error::SudachiError;
+bitflags! {
+    /// A set of categories for a character
+    ///
+    /// Implemented as a bitset with fixed size
+    #[repr(transparent)]
```

Comment:
What is this attribute for? (I'm just curious.)

Reply:
From the binary point of view treat it as the single internal field, so this struct and related things actually do not exist in the binary at all, it's only single u32 which is usually passed via registers and does not even go to the stack most of the time.

Reply:
In Rust this attribute can be assigned to any struct which has exactly one field.
https://doc.rust-lang.org/1.26.2/unstable-book/language-features/repr-transparent.html

Reply:
I got it, thank you